src/event/handlers/DiscordMessageLinkHandler.ts

```diff
@@ -8,15 +8,20 @@ class DiscordMessageLinkHandler extends EventHandler {
 	}
 
 	async handle(message: Message): Promise<void> {
-		const messagePreviewService = MessagePreviewService.getInstance();
 		const messageRegex = /https:\/\/(ptb\.)?discord(app)?\.com\/channels\//gm;
 
-		if (message.content.startsWith("!")) return;
-
 		if (message.content.match(messageRegex)) {
 			const linkIndex = message.content.search(messageRegex);
+			if (linkIndex == -1) return;
```

Comment:
This is the change i was suggesting, there's no need for `.match` as `.search` returns -1 if there was no match.

```suggestion
		const linkIndex = message.content.search(messageRegex);
		
		if (linkIndex !== -1) {
```